cmd/tsidp/tsidp.go

```diff
@@ -220,7 +350,24 @@ func (s *idpServer) authorize(w http.ResponseWriter, r *http.Request) {
 		clientID:    uq.Get("client_id"),
 	}
 
-	if r.URL.Path == "/authorize/localhost" {
+	// This URL is visited by the user who is being authenticated. If they are
```

Comment:
pre-existing but add a check that `redirect_uri` and `nonce` are non-empty?

Reply:
`state` too

Reply:
Oh i guess we should follow https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest and enforce that `redirect_uri` be specified. We should not pass empty `state` back if its not specified in the query like we do currently on line 387

Reply:
`nonce` is allowed to be absent per spec, we already don't report it if it's not provided (via `omitempty`). requiring `request_uri` and removing passing of empty state